src/audio_core/hle/wmf_decoder.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright 2018 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include "audio_core/hle/wmf_decoder.h"
+#include "audio_core/hle/wmf_decoder_utils.h"
+
+namespace AudioCore::HLE {
+
+class WMFDecoder::Impl {
+public:
+    explicit Impl(Memory::MemorySystem& memory);
+    ~Impl();
+    std::optional<BinaryResponse> ProcessRequest(const BinaryRequest& request);
+
+private:
+    std::optional<BinaryResponse> Initalize(const BinaryRequest& request);
+
+    void Clear();
+
+    std::optional<BinaryResponse> Decode(const BinaryRequest& request);
+
+    int DecodingLoop(ADTSData adts_header, std::array<std::vector<u8>, 2>& out_streams);
+
+    bool initalized = false;
+    bool selected = false;
+
+    Memory::MemorySystem& memory;
+
+    IMFTransform* transform = NULL;
```

Comment:
nullptr, This should probably be a unique_ptr with proper DTOR similar to what was done in the ffmpeg_decoder, to assure it always gets freed

Reply:
Not possible. WINAPIs do not conform to standard C++ RAII. Interfaces do not come with a destructor that frees resources. MS documents suggest `->Release()` is the only way to free the resources.

Reply:
Thats exactly what i meant inside the unique_ptr dtor the `->Release()` should get called